### DISCUSSION TITLE
GitHub failing to detect index.html.

### DISCUSSION THREAD

**Discussion by Disha-Sachdeva** [community/community#58178] [Other Feature Feedback, Questions, & Ideas] [answered]
### Select Topic Area

Bug

### Body

Hey there! I was doing my final assignment for a Coursera Web Development course and it's giving an error(404) even though I have explicitly defined index.html in my URL. All my other module assignments are working just fine. The existing solutions in the community did not work for me. I would really appreciate assistance on the issue. I am pasting the URL below for your reference. 

https://disha-sachdeva.github.io/my-coursera/module5_sol/index.html

For further reference, below is the link to my repository.

https://github.com/Disha-Sachdeva/my-coursera/tree/gh-pages

**Comment by the-officialjosh** [comment]
The error you encountered indicates that the project's build workflow stopped due to a missing component. Specifically, in the module5 section, you can see a red 'x' close to the file name in module_5 indicating an error in the workflow which terminated the build hence it never got to deploying your work. Clicking on it reveals the following error message:

"Error: fatal: No URL found for submodule path 'fullstack-course4' in .gitmodules
Error: The process '/usr/bin/git' failed with exit code 128."

To resolve this issue, I can help you investigate and fix the problem related to the 'fullstack-course4' submodule. Although I'm not familiar with the exact purpose of this submodule, To address this issue, you can follow these steps:

Open the .gitmodules file in your project.
Locate the entry related to the 'fullstack-course4' submodule.
Check if the submodule is required for your project. If it is not needed, you can remove the corresponding entry from the .gitmodules file.
If the submodule is necessary for your project, ensure that the submodule's URL is correctly specified in the .gitmodules file. Double-check for any typos or mistakes in the URL.
Save the changes to the .gitmodules file.
Commit and push the changes to your repository.

Once resolved, your project should build properly.

**Reply by Disha-Sachdeva** [reply]
Sir, I really appreciate your help. I checked both my master and gh-pages branches but there did not exist any .gitmodules file. Also, I am new to git and don't know much about it but I really need to submit my final assignment so I humbly request you to provide me with further assistance if you can.

**Reply by the-officialjosh** [reply]
I can't find your repo anymore did you delete your files? I was hoping to fork it and help you fix it. If you were able to fix it please keep me posted

**Reply by Disha-Sachdeva** [reply]
No, I didn't. If the same link isn't working then I'll paste it again. I, however, did delete fullstack-course4 folder since I didn't require it anymore.
 
Repo link :
https://github.com/Disha-Sachdeva/my-coursera/tree/gh-pages  

Site link:
https://disha-sachdeva.github.io/my-coursera/module5_sol/index.html

**Reply by the-officialjosh** [reply]
So From the build file i could spot another error but this time with your scss-   Conversion error: Jekyll::Converters::Scss encountered an error while converting 'assets/css/style.scss':
                    No such file or directory @ dir_chdir - /github/workspace/docs

The main problem is that with GHpages there is a testing engine automated into it so any issues with your code will stop it from being answered

**Reply by the-officialjosh** [reply]
Let me check your code and get back to you in a minute

**Reply by Disha-Sachdeva** [reply]
Sure, Sir. Thank you very much for your time and effort. I really appreciate it.

**Reply by the-officialjosh** [reply]
Wait have you pushed the project to the main branch? It seems you are working on a branch called gh-pages. The main was what I saw on clicking your link. Can you try merging your progress in gh-pages with main that could be the issue, I'll wait

**Reply by Disha-Sachdeva** [reply]
Sir, 
1. I tried merging but in my git bash, main did not exist but it did exist in my repo(on github.com) so I created a new 'main'. 
2. After I put in the merge command, when I tried to git add . , it is returning an infinite number of warnings and git commit is also resulting in warnings.

**Reply by Disha-Sachdeva** [reply]
![Screenshot (152)](https://github.com/community/community/assets/110038383/9b75cfd7-1337-43d1-bc12-8aaec32606df)

**Reply by the-officialjosh** [reply, marked as answer]
Considering your deadline how about you reinitialise the git. This time maybe you don't add jekyll. 

I noticed that you're experiencing difficulties with the Jekyll build on GitHub Pages, and I have a potential solution that might help resolve the issue. Please note that this solution involves starting afresh with the repository on GitHub. Here's what you can try:

1. Make sure you have a backup: Before proceeding, ensure that you have a local backup of your repository in case you need to refer back to any files or commits.

2. Delete the `.git` folder: On your local machine, navigate to the root directory of your project and locate the hidden `.git` folder. Delete this folder, as it contains the version control information for your repository.

3. Reinitialize the repository: Open a terminal or command prompt in the root directory of your project and run the following commands:

```
git init
git add .
git commit -m "Initial commit"
```

These commands will reinitialize your repository as a new Git repository, add all files to the repository, and create an initial commit.

4. Create a new repository on GitHub: On the GitHub website, create a new repository with a different name or in a different location than the previous one. You can follow the usual steps to create a new repository and obtain the repository URL.

5. Link your local repository to the new GitHub repository: Back in your terminal or command prompt, run the following command, replacing `[new-repository-url]` with the URL of your new repository:

```
git remote add origin [new-repository-url]
```

This command will link your local repository to the new GitHub repository.

6. Push your code: Finally, execute the following command to push your code to the new repository:

```
git push -u origin master
```

This command will upload your code to the new repository on GitHub.

The you can add the gitub pages again

By following these steps, you'll essentially be starting fresh with a new Git repository on GitHub. Hopefully, this will help resolve the issue you were facing with the Jekyll build on GitHub Pages. Feel free to reach out if you need any further assistance or clarification.

**Reply by Disha-Sachdeva** [reply]
It finally worked. Thank you so much, Sir. You don't know how much this means to me. Once again thank you for your time and assistance.

**Reply by the-officialjosh** [reply]
You're welcome. But remember this is not always the go-to option, especially in the future when you are working professionally bt glad that this was able to fix it so you could submit your work

**Reply by Disha-Sachdeva** [reply]
I will surely remember this. Thanks for the guidance.
